packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1332,7 +1336,7 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
   TextPosition? _textSelectionStart;
   TextPosition? _textSelectionEnd;
 
-  bool _selectableContainsOriginWord = false;
+  bool _selectableContainsOriginTextBoundary = false;
```

Comment:
maybe rename this to _selectionStartedInHere and add some comments to explain why it needs to keep track of this.